lib/resty/core/regex.lua

```diff
@@ -153,6 +154,14 @@ local function get_max_regex_cache_size()
     return max_regex_cache_size
 end
 
+function _M.regex_cache_is_empty()
+    if not regex_match_cache then
+        return true
+    else
+        return next(regex_match_cache.hasht) == nil
```

Comment:
@alubbe I think it's better to just keep the track of regex caching ourselves via a top-level local flag variable.